media_driver/linux/common/os/i915/mos_bufmgr.c

```diff
@@ -1559,7 +1677,41 @@ map_wc(struct mos_linux_bo *bo)
         return -EINVAL;
 
     /* Get a mapping of the buffer if we haven't before. */
-    if (bo_gem->mem_wc_virtual == nullptr) {
+    if (bo_gem->mem_wc_virtual == nullptr && bufmgr_gem->has_lmem) {
+        struct drm_i915_gem_mmap_offset mmap_arg;
+
+        MOS_DBG("bo_map_wc: mmap_offset %d (%s), map_count=%d\n",
+            bo_gem->gem_handle, bo_gem->name, bo_gem->map_count);
+
+        memclear(mmap_arg);
+        mmap_arg.handle = bo_gem->gem_handle;
+        /* To indicate the uncached virtual mapping to KMD */
+        mmap_arg.flags = I915_MMAP_OFFSET_FIXED;
```

Comment:
Another difference vs. i915_production/mos_bufmgr.c.

Pay attention on the usage of I915_MMAP_OFFSET_FIXED in all calls to DRM_IOCTL_I915_GEM_MMAP_OFFSET ioctl.

See https://github.com/freedesktop/drm-tip/blob/85801dbe86b9dc8eae00dd10347b7f3552f9bef9/include/uapi/drm/i915_drm.h#L897:
```
* On devices with local memory `I915_MMAP_OFFSET_FIXED` is the only valid
* type. On devices without local memory, this caching mode is invalid.
```